assets/js/base/context/hooks/use-validation.ts

```diff
@@ -0,0 +1,54 @@
+/**
+ * External dependencies
+ */
+import { useCallback } from '@wordpress/element';
+import type {
+	ValidationData,
+	ValidationContextError,
+} from '@woocommerce/type-defs/contexts';
+
+/**
+ * Internal dependencies
+ */
+import { useValidationContext } from '../providers/validation/';
+
+/**
+ * Custom hook for setting for adding errors to the validation system.
+ */
+export const useValidation = (): ( (
+	validationErrorId: string
+) => ValidationData ) => {
+	const {
+		hasValidationErrors,
+		getValidationError,
+		clearValidationError,
+		hideValidationError,
+		setValidationErrors,
+	} = useValidationContext();
```

Comment:
I'm fine with this because it exposes what we have. Right now we put the onus on components to implement this, and then expose their validation errors within the context provider. That works fine.

I do however wonder if we should only pass down the current status and let components manage their own validation error + states, with an event or callback to block checkout and show a blocking error message. I don't know if this is something being looked at separately so shouldn't stop this being merged if we just want to roll with the existing system. Did you have any thoughts on this?

Reply:
We're going to revisit the whole validation system at some point soon. We still need some way to tell checkout and blocks:
- There's an error.
- The error id and message.
- Various states.

I'm not sure what will come out of that refactor, but we still need a place to collect all existing errors.